integration/rpc/sync/sync_agent_client.py

```diff
@@ -0,0 +1,50 @@
+import grpc
+
+import common_pb2
+import syncagent_pb2
+import syncagent_pb2_grpc
+from google.protobuf import empty_pb2
+
+from common.interceptor import IdentityValidationInterceptor
+
+
+class SyncAgentClient(object):
+    def __init__(self, address, volume_name=None, instance_name=None):
+        self.address = address
+        channel = grpc.insecure_channel(address)
+        # Default volume_name = instance_name = None disables identity
+        # validation.
+        interceptor = IdentityValidationInterceptor(volume_name, instance_name)
+        self.channel = grpc.intercept_channel(channel, interceptor)
+        self.stub = syncagent_pb2_grpc.SyncAgentServiceStub(self.channel)
+
+    # Only two client methods have been implemented in service of 
+    # test_validation_fails_with_client. Additional methods are TODO.
+
+    def replica_rebuild_status(self):
+        return self.stub.ReplicaRebuildStatus(empty_pb2.Empty())
+    
+    def sync_files(self, from_address, sync_file_info_tuples, fast_sync,
+                   file_sync_http_client_timeout):
+
+        sync_file_info_list = []
+        for tuple in sync_file_info_tuples:
+            info = common_pb2.SyncFileInfo(from_file_name="some_name",
```

Comment:
We don't even bother looking at the tuple, which could be anything, but we make a hard-coded info - the same one for each item in the array of tuples, if there are more than one.

Reply:
This looks weird. Not sure why this API implementation ignores that content of input `sync_file_info_tuples` and directly generates a hard-coded list for `self.stub.FilesSync()` invocation.

Reply:
Good catch. There was some weirdness around imports (only on ARM) that caused me to pass this info to sync_files as a tuple, but I intended to actually use the info (no hard code it)! Didn't notice because the whole point of my new test is to fail without syncing a file. Fixing.